core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -60,15 +60,15 @@ class LogManager(logDirs: Seq[File],
                  val retentionCheckMs: Long,
                  val maxPidExpirationMs: Int,
                  scheduler: Scheduler,
-                 val brokerState: AtomicReference[BrokerState],
+                 val cleanShutdownFunc: Boolean => Unit, // we send true if the shutdown was clean, otherwise false
```

Comment:
I think if you want to go this route you should make this a `CompletableFuture[Boolean]`, and remove `BrokerState` completely from `LogManager.scala`.

Reply:
>  think if you want to go this route you should make this a CompletableFuture[Boolean], and remove BrokerState completely from LogManager.scala.

Thanks, that simplified things.